csharp/src/Drivers/BigQuery/BigQueryStatement.cs

```diff
@@ -0,0 +1,422 @@
+            {
+                Date64Array date64Array = (Date64Array)arrowArray;
+
+                return date64Array.GetDateTime(index);
+            }
+            else if (arrowArray is Time64Array)
+            {
+                return ((Time64Array)arrowArray).GetValue(index);
+            }
+            else if (arrowArray is TimestampArray)
+            {
+                TimestampArray timestampArray = (TimestampArray)arrowArray;
+                DateTimeOffset dateTimeOffset = timestampArray.GetTimestamp(index).Value;
+                return dateTimeOffset;
+            }
+            else if (arrowArray is StructArray)
+            {
+                StructArray structArray = (StructArray)arrowArray;
+                return SerializeToJson(structArray, index);
+            }
+            // maybe not be needed?
+            else if (arrowArray is ListArray)
+            {
+                return ((ListArray)arrowArray).GetSlicedValues(index);
+            }
+
+            return null;
+        }
+
+        static Schema TranslateSchema(TableSchema schema)
+        {
+            return new Schema(schema.Fields.Select(TranslateField), null);
+        }
+
+        static Field TranslateField(TableFieldSchema field)
+        {
+            return new Field(field.Name, TranslateType(field), field.Mode == "NULLABLE");
+        }
+
+        static IArrowType TranslateType(TableFieldSchema field)
+        {
+            // per https://developers.google.com/resources/api-libraries/documentation/bigquery/v2/java/latest/com/google/api/services/bigquery/model/TableFieldSchema.html#getType--
+
+            switch (field.Type)
+            {
+                case "INTEGER" or "INT64":
+                    return Int64Type.Default;
+                case "FLOAT" or "FLOAT64":
+                    return DoubleType.Default;
+                case "BOOL" or "BOOLEAN":
+                    return BooleanType.Default;
+                case "STRING":
+                    return StringType.Default;
+                case "BYTES":
+                    return BinaryType.Default;
+                case "DATETIME":
+                    return TimestampType.Default;
+                case "TIMESTAMP":
+                    return TimestampType.Default;
+                case "TIME":
+                    return Time64Type.Default;
+                case "DATE":
+                    return Date64Type.Default;
+                case "RECORD" or "STRUCT":
+                    // its a json string
+                    return StringType.Default;
+                // get schema cannot get precison and scale
+                case "NUMERIC" or "DECIMAL":
+                    return new Decimal128Type(38, 9);
+                //(int)field.Precision, (int)field.Scale);
+
+                // treat these values as strings
+                case "BIGNUMERIC" or "BIGDECIMAL" or "GEOGRAPHY":
+                    return StringType.Default;
```

Comment:
Maybe to avoid having a breaking change in the future, this behavior could be driven by an option which says "report large decimals as strings?"

Reply:
fixed in latest push